deps/v8/src/execution/isolate.cc

```diff
@@ -604,6 +604,48 @@ StackTraceFailureMessage::StackTraceFailureMessage(Isolate* isolate, void* ptr1,
   }
 }
 
+static std::string ToStdString(int n) {
+  char buf[20];
+  snprintf(buf, sizeof(buf), "%d", n);
+  return std::string(buf);
+}
+
+static std::string ToStdString(Handle<Object> obj) {
+  if (obj->IsString()) {
+    std::unique_ptr<char[]> contents = String::cast(*obj).ToCString();
+    return std::string(contents.get());
+  }
+  return std::string("<object>");
+}
+
+// Some internal frames appear non-deterministically in stack traces. Normally
+// these aren't visible to content scripts, but they can still be observed by
+// overriding Error.prepareStackTrace. Filter these out so that scripts which
+// observe the stack this way still behave deterministically.
+static bool RecordReplayIgnoreStackFrame(Handle<StackTraceFrame> frame) {
+  if (!recordreplay::IsRecordingOrReplaying()) {
+    return false;
+  }
+
+  static std::pair<const char*, const char*> ignoredFrames[] = {
+    { "<object>", "runMicrotasks" },
```

Comment:
Curious, when does `StackTraceFrame::GetFileName(frame)` return an object?

Reply:
I'm not sure it's an object, it might be an undefined value instead.